cdap-ui/app/cdap/components/ConfigurationGroup/index.tsx

```diff
@@ -80,10 +102,30 @@ const ConfigurationGroupView: React.FC<IConfigurationGroupProps> = ({
       };
 
       changeParentHandler(newValues);
+      referenceValueForUnMount.current = {
+        configurationGroups: filteredConfigurationGroups,
+        values: newValues,
+      };
     },
-    [widgetJson, pluginProperties]
+    [values]
   );
 
+  // This onUnMount is to make sure we clear out all properties that are hidden.
+  React.useEffect(() => {
+    return () => {
+      const newValues = { ...referenceValueForUnMount.current.values };
+      const configurationGroups = referenceValueForUnMount.current.configurationGroups;
+      configurationGroups.forEach((group) => {
+        group.properties.forEach((property) => {
+          if (property.show === false) {
+            console.log('deleting property: ', property.name);
+            delete newValues[property.name];
```

Comment:
Removing the properties only during unmount might be a problem for stage level validation. The stage level validation happens before the unmount

Reply:
While doing stage level validation this needs to be done again.

Right now I am not deleting the properties as soon as the filter condition becomes false as the user might accidentally change something that will cause it to become false. If we delete it right away that means the user has to type in everything again. 

This is so that the properties are removed only when the user closes the modal instead.